model/__init__.py

```diff
@@ -659,13 +659,13 @@ class Patron(Base):
     _synchronize_annotations = Column(Boolean, default=None,
                                       name="synchronize_annotations")
 
-    loans = relationship('Loan', backref='patron')
-    holds = relationship('Hold', backref='patron')
+    loans = relationship('Loan', backref='patron', cascade='delete')
+    holds = relationship('Hold', backref='patron', cascade='delete')
 
-    annotations = relationship('Annotation', backref='patron', order_by="desc(Annotation.timestamp)")
+    annotations = relationship('Annotation', backref='patron', order_by="desc(Annotation.timestamp)", cascade='delete')
 
     # One Patron can have many associated Credentials.
-    credentials = relationship("Credential", backref="patron")
+    credentials = relationship("Credential", backref="patron", cascade="delete")
```

Comment:
I checked that this is right, and I think you could slightly improve these by changing to `cascade="delete-orphan"`. This will delete (e.g.) a `Loan` if its `patron` is set to None. (Which shouldn't happen the way we wrote the code.)

Reply:
Adriana and I looked into this and there are tests that assume you can set `Loan.patron=None` without deleting the loan, so we decided the minimal benefit wasn't worth changing this.

Reply:
I think for ODL there are in fact Loans with no patron.